packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2155,7 +2155,7 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
         child: AnimatedDefaultTextStyle(
           duration:_kTransitionDuration,
           curve: _kTransitionCurve,
-          style: widget._labelShouldWithdraw
+          style: widget._labelShouldWithdraw || decoration.floatingLabelBehavior == FloatingLabelBehavior.always
```

Comment:
I wonder if `_labelShouldWithdraw` should just take into account `decoration.floatingLabelBehavior == FloatingLabelBehavior.always`. I looked through the usages in this file and it is almost always used along with `decoration.floatingLabelBehavior == FloatingLabelBehavior.always` so it seems reasonable. What do you think?

Reply:
Yeah I think it makes sense to change this even in the semantics case, since its depending on whether the label is floating. I'll keep an eye out if this causes any issues.